exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -0,0 +1,249 @@
+import io.opencensus.common.Functions;
+import io.opencensus.common.OpenCensusLibraryInformation;
+import io.opencensus.common.Timestamp;
+import io.opencensus.trace.Annotation;
+import io.opencensus.trace.NetworkEvent;
+import io.opencensus.trace.NetworkEvent.Type;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanData.TimedEvent;
+import io.opencensus.trace.export.SpanData.TimedEvents;
+import io.opencensus.trace.export.SpanExporter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/** Exporter to Stackdriver Trace API v2. */
+final class StackdriverV2ExporterHandler extends SpanExporter.Handler {
+  private static final String AGENT_LABEL_KEY = "g.co/agent";
+  private static final String AGENT_LABEL_VALUE_STRING =
+      "opencensus-java [" + OpenCensusLibraryInformation.VERSION + "]";
+  private static final AttributeValue AGENT_LABEL_VALUE =
+      AttributeValue.newBuilder()
+          .setStringValue(toTruncatableStringProto(AGENT_LABEL_VALUE_STRING))
+          .build();
+
+  private final String projectId;
+  private final TraceServiceClient traceServiceClient;
+  private final ProjectName projectName;
+
+  private StackdriverV2ExporterHandler(String projectId, TraceServiceClient traceServiceClient) {
+    this.projectId = checkNotNull(projectId, "projectId");
+    this.traceServiceClient = traceServiceClient;
+    projectName = ProjectName.newBuilder().setProject(projectId).build();
+  }
+
+  static StackdriverV2ExporterHandler createWithCredentials(
+      Credentials credentials, String projectId) throws IOException {
+    checkNotNull(credentials, "credentials");
+    TraceServiceSettings traceServiceSettings =
+        TraceServiceSettings.newBuilder()
+            .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+            .build();
+    return new StackdriverV2ExporterHandler(
+        projectId, TraceServiceClient.create(traceServiceSettings));
+  }
+
+  static StackdriverV2ExporterHandler create(String projectId) throws IOException {
+    return new StackdriverV2ExporterHandler(projectId, TraceServiceClient.create());
+  }
+
+  // TODO(bdrutu): Add tests for this.
+  private Span generateSpan(SpanData spanData) {
+    SpanContext context = spanData.getContext();
+    final String traceIdHex = encodeTraceId(context.getTraceId());
+    final String spanIdHex = encodeSpanId(context.getSpanId());
+    SpanName spanName =
+        SpanName.newBuilder().setProject(projectId).setTrace(traceIdHex).setSpan(spanIdHex).build();
+    Span.Builder spanBuilder =
+        Span.newBuilder()
+            .setNameWithSpanName(spanName)
+            .setSpanId(encodeSpanId(context.getSpanId()))
+            .setDisplayName(toTruncatableStringProto(spanData.getName()))
```

Comment:
Truncate to 128.

Reply:
https://github.com/census-instrumentation/opencensus-java/issues/813